app/src/main/java/com/edu/uni/augsburg/uniatron/stepcounter/StepCountService.java

```diff
@@ -0,0 +1,97 @@
+package com.edu.uni.augsburg.uniatron.stepcounter;
+
+import android.app.Service;
+import android.content.Intent;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.AsyncTask;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+//import com.edu.uni.augsburg.uniatron.MainApplication;
+
+/**
+ * The step count service collects steps and commits them to the database.
+ *
+ * @author Leon Wöhrl
+ */
+public class StepCountService extends Service implements SensorEventListener {
+
+    private SensorManager sensorManager;
+    private Sensor stepDetectorSensor;
+    private static final int COMMIT_SIZE = 10;
+    private int currentSteps;
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        currentSteps = 0;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent,   int flags, int startId) {
+        // grab step detector and register the listener
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+        stepDetectorSensor = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_DETECTOR);
+        sensorManager.registerListener(this, stepDetectorSensor, SensorManager.SENSOR_DELAY_FASTEST);
+
+        // this causes the OS to restart the service if it has been force stopped
+        return START_STICKY;
+    }
+
+    @Override
+    public void onSensorChanged(SensorEvent event) {
+        // detects every single step
+        if (event.sensor.getType() == Sensor.TYPE_STEP_DETECTOR) {
+            int detectedSteps = (int) event.values[0];
```

Comment:
Does this always return a value or could this also become null?
```java
event.values[0]
```
If event.values is null, then this would produce a NullPointerException. If the event.values is empty, then a IndexOutOfBoundException would be thrown. Is this a possible scenario?

Reply:
should never happen, as onSensorChanged only triggers when values are present. I'll include a try/catch anyways, and execute nothing should such event occur